### DISCUSSION TITLE
Help: Publishing via C#

### DISCUSSION THREAD

**Discussion by ksmuthucv** [bluenviron/mediamtx#2200] [Questions] [answered]
Hello Team,
I’m trying to publish raw nals of a frame (h264/h265) to the mediamtx server using C#. Can you please advise what must be done to publish, successfully? Below is what i’m trying to do as of now.

1. Get the vps (h265 only), sps and pps.
2. Connect to the server.
3. Send sdp as part of ANNOUNCE to the server. (would assume, similar to client-publish-format-h265/main.go, #40 media.Media)
4. Start sending the entire frame (h264/h265) nals (byte[]) to the server (not sure any fragmentation/modification is required before sending).

URL for publishing: rtsp://localhost:8554/stream

**Output from mediamtx server:**
2023/08/13 17:40:28 INF MediaMTX v1.0.0
2023/08/13 17:40:28 INF [RTSP] listener opened on :8554 (TCP), :8000 (UDP/RTP), :8001 (UDP/RTCP)
2023/08/13 17:40:28 INF [RTMP] listener opened on :1935
2023/08/13 17:40:28 INF [HLS] listener opened on :8888
2023/08/13 17:40:28 INF [WebRTC] listener opened on :8889 (HTTP)
2023/08/13 17:40:28 INF [SRT] listener opened on :8890 (UDP)
2023/08/13 17:40:46 INF [RTSP] [conn [::1]:60348] opened
2023/08/13 17:40:46 INF [RTSP] [session ca6f7500] created by [::1]:60348
2023/08/13 17:40:46 INF [RTSP] [conn [::1]:60348] closed (no one is publishing to path 'stream')
2023/08/13 17:40:46 INF [RTSP] [session ca6f7500] destroyed (not in use)
2023/08/13 17:40:46 INF [RTSP] [conn [::1]:60349] opened
2023/08/13 17:40:46 INF [RTSP] [conn [::1]:60349] closed (buffer length exceeds 64)
2023/08/13 17:40:46 INF [RTSP] [conn [::1]:60350] opened
2023/08/13 17:40:46 INF [RTSP] [conn [::1]:60350] closed (buffer length exceeds 64)

Thanks,
Muthu

**Comment by aler9** [comment, marked as answer]
Hello, my advice is to set `logLevel: debug` inside the server configuration. This allows to view RTSP requests and responses in the console. Then try to publish with FFmpeg. Then try to publish with your C# client. You'll be able to clearly find the differences.

**Comment by DivekarNikhil** [comment]
What is used in real life application of c#

**Comment by ksmuthucv** [comment]
Was able to solve the publishing with help of debug comments! Thanks @aler9

[a comment by an automated account is not shown]
